app/src/main/java/me/hufman/androidautoidrive/music/controllers/SpotifyAppController.kt

```diff
@@ -303,7 +368,7 @@ class SpotifyAppController(context: Context, val remote: SpotifyAppRemote, val w
 			val item = recentlyPlayed?.items?.get(0)
 			if (item != null) {
 				remote.imagesApi.getImage(item.imageUri, Image.Dimension.THUMBNAIL).setResultCallback { coverArt ->
-					queueMetadata = QueueMetadata(item.title, item.subtitle, queueItems, coverArt, item.uri)
+					queueMetadata?.coverArt = coverArt
```

Comment:
What does this change improve? Are there places in the code that hold a reference to the queueMetadata that need the new coverart? If it's just to reuse the fake list of queueItems, it can probably be fetched from `QueueMetadata(.., .., queueMetadata.queueItems, ..)`

Reply:
The reasoning for this change was I was looking at this more in depth as part of this task and I noticed we are creating a brand new QueueMetadata object just to introduce a cover art bitmap not only adds extra overhead in creating a new instance of the queueMetadata (yes it's a very small maybe immeasurable impact 😄) but also seems unnecessary and not as clean as simply updating the cover art on the existing queueMetadata instance. This way we only have a single queueMetadata instance (during the queue "session") that is being updated as the data is becoming available for it

Reply:
Good thinking, for considering performance on phones! However, the new queueMetadata objects should be relatively rare, though I do intend to run a real performance analysis to know for certain where any hot spots are.
Some places in the code (such as `EnqueuedView.redraw()`) hold a reference to the previous version of queueMetadata, checking object equality to know when the controller has a new queueMetadata to show. Simply updating the existing object won't notify the change properly.

Reply:
Ah I see what you mean and digging further into Kotlin data classes it turns out that they automatically implement the equals(..) method so all of those references to the QueueMetadata won't be doing object equality checks on the address of the object but will do member variable equality which then means that we won't have to worry about those equality checks not returning correctly for those scenarios (docs for reference: https://kotlinlang.org/docs/data-classes.html)

Reply:
This is true, indeed! However, the EnqueuedView holds a reference to SpotifyAppController's queueMetadata, and so updating this queueMetadata object will also mean that the EnqueuedView's queueMetadata will immediately have this new coverart, which means its redraw logic won't notice the update and any deferred coverart loading won't display until the next time the EnqueuedView is opened.
I'd prefer to leave the QueueMetadata class as immutable to reduce the possible side effects.

Reply:
Ohh I think I'm understanding now, I agree it is better to leave the QueueMetadata class as immutable then. I reverted the QueueMetadata to be immutable and reverted this logic

Reply:
It looks like my comment was deleted somehow, I think when my session glitched and I went to remove the extra comment I accidentally deleted this one as well. 

I reverted the QueueMetadata modifications and the code to set the coverArt to how it was before